x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/StringUtils.java

```diff
@@ -157,7 +157,7 @@ public static String wildcardToJavaPattern(String pattern, char escape) {
                     case '*' -> regex.append(escaped ? "\\*" : ".*");
                     case '?' -> regex.append(escaped ? "\\?" : ".");
                     default -> {
-                        if (escaped) {
+                        if (escaped && escape != curr) {
                             throw new QlIllegalArgumentException(
                                 "Invalid sequence - escape character is not followed by special wildcard char"
                             );
```

Comment:
out-of-scope: We could add to the comment on top that using `*` or `?` as escape char is not advised because this will make it impossible to input a literal `*` or `?` - and turn the comment into a javadoc.